scene/gui/text_edit.cpp

```diff
@@ -1372,13 +1373,19 @@ void TextEdit::_notification(int p_what) {
 
 				if (completion_options.size() < 50) {
 					for (int i = 0; i < completion_options.size(); i++) {
-						int w2 = MIN(cache.font->get_string_size(completion_options[i]).x, cmax_width);
+						int w2 = MIN(cache.font->get_string_size(completion_options[i].display).x, cmax_width);
 						if (w2 > w)
 							w = w2;
 					}
 				} else {
 					w = cmax_width;
 				}
+#ifdef TOOLS_ENABLED
+				Size2 icon_area_size(get_row_height(), get_row_height());
+				if (Engine::get_singleton()->is_editor_hint()) {
```

Comment:
There shouldn't be a need to check for tools or editor, can increase the width if there is an icon in the completion options.

Reply:
I don't want to add `TOOLS_ENABLED` and `is_editor_hint` check in the scene part code and I try to avoid include editor headers here. But I didn't find a better way to implement this feature.
And there are already so many `TOOLS_ENABLED` check in `text_edit.cpp`.
I'd like to known a better solution for this.

Reply:
I think that now that the icon is in the `ScriptCodeCompletionOption`, you can use that to see if the width should be increased, without having to use `TOOLS_ENABLED`.

@reduz told us that we need to remove all `scene/` code that has dependencies on `editor/`: #29730. Though in this case `TOOLS_ENABLED` only without using editor-specific headers is fine, but as mentioned above the relevant information should already be in the completion option.